commit 3374830a6a0856afd724f96a7795b2470e22adc3 (refs/changes/23/1423/3)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-02-06T20:20:49+01:00 (2 years, 8 months ago)
    
    TOOLS-1667 sdcadm: Move `experimental cns` to `post-setup cns`
    Trent Mick <trent.mick@joyent.com>

diff --git a/CHANGES.md b/CHANGES.md
index 73d86eb..c86e8d1 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -10,6 +10,11 @@
 
 # sdcadm Changelog
 
+## 1.15.4
+
+- Deprecate 'sdcadm experimental cns' in favor of 'sdcdm post-setup cns'
+
+
 ## 1.15.3
 
 - Deprecate 'sdcadm experimental add-new-agent-svcs' and roll its functionality
diff --git a/lib/cli/experimental.js b/lib/cli/experimental.js
index 770e69d..acdb5f8 100644
--- a/lib/cli/experimental.js
+++ b/lib/cli/experimental.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright 2016, Joyent, Inc.
+ * Copyright 2017 Joyent, Inc.
  */
 
 /*
@@ -77,8 +77,8 @@ require('./do_update_docker').do_update_docker;
 ExperimentalCLI.prototype.do_install_docker_cert =
 require('./do_install_docker_cert').do_install_docker_cert;
 
-ExperimentalCLI.prototype.do_cns =
-require('./do_cns').do_cns;
+// Deprecated: TOOLS-1667
+ExperimentalCLI.prototype.do_cns = require('../post-setup/cns').do_cns;
 
 
 
diff --git a/lib/cli/do_cns.js b/lib/post-setup/cns.js
similarity index 96%
rename from lib/cli/do_cns.js
rename to lib/post-setup/cns.js
index 2af875c..a9eff6d 100644
--- a/lib/cli/do_cns.js
+++ b/lib/post-setup/cns.js
@@ -5,11 +5,11 @@
  */
 
 /*
- * Copyright (c) 2015, Joyent, Inc.
+ * Copyright 2017 Joyent, Inc.
  */
 
 /*
- * The 'sdcadm experimental cns' CLI subcommand.
+ * The 'sdcadm post-setup cns' CLI subcommand.
  */
 
 var util = require('util'),
@@ -19,7 +19,7 @@ var vasync = require('vasync');
 var errors = require('../errors');
 var DownloadImages = require('../procedures/download-images').DownloadImages;
 var shared = require('../procedures/shared');
-
+var common = require('../common');
 
 function do_cns(subcmd, opts, args, cb) {
     var self = this;
@@ -30,6 +30,11 @@ function do_cns(subcmd, opts, args, cb) {
         return cb(new errors.UsageError('too many args: ' + args));
     }
 
+    if (self.name === 'sdcadm experimental') {
+        self.progress('Warning: `sdcadm experimental cns` is deprecated.\n' +
+            common.indent('Please use `sdcadm post-setup cns` instead.\n'));
+    }
+
     var start = Date.now();
     var svcData = {
         name: 'cns',
diff --git a/lib/post-setup/index.js b/lib/post-setup/index.js
index 6879e42..443fca0 100644
--- a/lib/post-setup/index.js
+++ b/lib/post-setup/index.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2016, Joyent, Inc.
+ * Copyright 2017 Joyent, Inc.
  */
 
 /*
@@ -73,6 +73,7 @@ PostSetupCLI.prototype.do_dev_sample_data =
 
 PostSetupCLI.prototype.do_docker = require('./docker').do_docker;
 PostSetupCLI.prototype.do_cmon = require('./cmon').do_cmon;
+PostSetupCLI.prototype.do_cns = require('./cns').do_cns;
 
 //---- exports
 
diff --git a/man/man1/sdcadm.1.ronn b/man/man1/sdcadm.1.ronn
index 101281e..58631c6 100644
--- a/man/man1/sdcadm.1.ronn
+++ b/man/man1/sdcadm.1.ronn
@@ -509,6 +509,14 @@ and create an initial instance of "cmon" on the headnode and
     be setup. If not specified, then cmon-agent will be setup on all setup
     servers.
 
+### sdcadm post-setup cns
+
+Create the "cns" service and a first instance.
+
+`-h, --help`
+    Show this help.
+
+
 ### sdcadm post-setup dev-headnode-prov
 
 Make the headnode provisionable, for development and testing.
diff --git a/package.json b/package.json
index a79a142..3f15cbe 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "sdcadm",
   "description": "Administer a SmartDataCenter (SDC) standup",
-  "version": "1.15.3",
+  "version": "1.15.4",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
